src/lib/tuning/abstract_selector.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "tuning/tuning_choice.hpp"
+#include "tuning/tuning_operation.hpp"
+
+namespace opossum {
+
+/**
+ * An AbstractSelector transforms an unordered list of TuningChoices into
+ * a concrete sequence of TuningOperations.
+ *
+ * It considers a cost budget that the entire operation sequence as well as any
+ * continuous subsequence from the beginning must not exceed.
+ * The operation sequence is prioritized by expected performance impact on the system,
+ * so that the most beneficial operations come before less useful operations.
+ *
+ * The underlying problem relates to the knapsack problem, but goes beyond that
+ * in scope (e.g. choices invalidating other choices).
+ * For more information, see e.g. Wikipedia:
+ *   https://en.wikipedia.org/wiki/Knapsack_problem
+ */
+class AbstractSelector {
+ public:
+  virtual ~AbstractSelector() {}
+  /**
+   * Determine the index operation sequence as specified above based on the
```

Comment:
This is the abstract selector. Why are you talking about an index here?

Reply:
true, done